### PR TITLE
Deleted dataset UI support

### LINKED ISSUE #2148: Add UI support for deleted tables

### PR DESCRIPTION
**Pull request by darth-coder00**
### Describe your changes :
Closes #2148 

I worked on the Deleted dataset UI

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
<p align="center">
Screenshot 2022-01-22 at 8 43 24 PM

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t @Sachin-chaurasiya

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx [hunk 1]
```diff
@@ -214,10 +226,13 @@ const EntityPageInfo = ({
...
                     className={classNames(
                       'tw-text-xs tw-border-r tw-font-medium tw-py-1 tw-px-2 tw-rounded-l focus:tw-outline-none',
-                      isFollowing ? 'tw-border-white' : 'tw-border-primary'
+                      isFollowing ? 'tw-border-white' : 'tw-border-primary',
+                      { 'tw-cursor-not-allowed': deleted }
                     )}
                     data-testid="follow-button"
-                    onClick={followHandler}>
+                    onClick={() => {
+                      !deleted && followHandler && followHandler();
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:234]:
```suggestion
                      !deleted && followHandler?.();
```
optional chaining makes calling optional functions easier than nullish coalescing.

File: openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx [hunk 2]
```diff
@@ -130,6 +132,14 @@ const TableDataCard: FunctionComponent<Props> = ({
               {stringToHTML(name)}
             </button>
           </h6>
+          {deleted && (
+            <>
+              <div className="tw-rounded tw-bg-error-lite tw-text-error tw-text-xs tw-font-medium tw-h-5 tw-px-1.5 tw-py-0.5 tw-ml-2">
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx:137]: We should make it a separate component if we are using it in more than one place

File: openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx [hunk 3]
```diff
@@ -114,7 +116,7 @@ const TableDataCard: FunctionComponent<Props> = ({
       data-testid="table-data-card"
       id={id}>
       <div>
-        <div className="tw-flex">
+        <div className="tw-flex tw-items-center">
           {/* {getEntityIcon(indexType)} */}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx:120]: Remove the commented code if not required.

File: openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx (hunk 2 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx:137]: UI might be changed so we can do it once it's final

File: openmetadata-ui/src/main/resources/ui/src/pages/DatasetDetailsPage/DatasetDetailsPage.component.tsx [hunk 4]
```diff
@@ -276,6 +277,16 @@ const DatasetDetailsPage: FunctionComponent = () => {
       });
   };
 
+  const getLineageData = () => {
+    getLineageByFQN(tableFQN, EntityType.TABLE)
+      .then((res: AxiosResponse) => {
+        setEntityLineage(res.data);
+      })
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatasetDetailsPage/DatasetDetailsPage.component.tsx:284]: we should add the catch block for error handling.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Minor changes are required. Otherwise LGTM.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**